doc/source/howtos/crash-recovery.rst

```diff
@@ -0,0 +1,295 @@
+file and start the `mysql` service:
+
+.. code-block:: bash
+
+   $ service mysql start --wsrep_sst_donor=nodeC
+
+.. _pxc-crash-recovery-scenario-3:
+
+Scenario: All three nodes are gracefully stopped
+================================================================================
+
+The cluster is deformed and the problem is to initialize it again. It is
+important that a PXC node writes its last executed position to the
+:file:`grastate.dat` file.
+
+By comparing the seqno number in this file, you can see which is the most
+advanced node (most likely the last stopped). The cluster must be bootstrapped
+using this node, otherwise nodes that had a more advanced position will have to
+perform the full :term:`SST` to join the cluster initialized from the less
+advanced one. As a result, some transactions will be lost). To bootstrap the
+first node, invoke the startup script like this:
+
+.. code-block::
+
+   $ service mysql bootstrap-pxc
+    
+.. note::
+
+   Even though you bootstrap from the most advanced node, the other
+   nodes have a lower sequence number. They will still have to join via the full :term:`SST`
+   because the *Galera Cache* is not retained on restart.
+
+   For this reason, it is recommended to stop writes to the cluster *before* its
+   full shutdown, so that all nodes can stop at the same position. See also
+   :variable:`pc.recovery`.
+
+.. _pxc-crash-recovery-scenario-4:
+
+Scenario: One node disappears from the cluster
+================================================================================
+
+This is the case when one node becomes unavailable due to power outage, hardware
+failure, kernel panic, mysqld crash, :command:`kill -9` on mysqld pid, etc.
+
+Two remaining nodes notice the connection to node A is down and start trying to
+re-connect to it. After several timeouts, node A is removed from the
+cluster. The quorum is saved (2 out of 3 nodes are up), so no service disruption
+happens. After it is restarted, node A joins automatically (as described in
+:ref:`pxc-crash-recovery-scenario-1`).
+
+.. _pxc-crash-recovery-scenario-5:
+
+Scenario: Two nodes disappear from the cluster
+================================================================================
+
+Two nodes are not available and the remaining node (node C) is not able to form
+the quorum alone. The cluster has to switch to a non-primary mode, where MySQL
+refuses to serve any SQL queries. In this state, the :command:`mysqld` process
+on node C is still running and can be connected to but any statement related to
+data fails with an error
+
+.. code-block:: guess
+
+   mysql> select * from test.t1;
+   ERROR 1047 (08S01): Unknown command
+
+Reads are possible until node C decides that it cannot access node A and
+node B. New writes are forbidden immediately thanks to the certification based
+replication in Galera. The log of node C reports the following
+
+.. code-block:: text
+
+   140814 0:42:13 [Note] WSREP: commit failed for reason: 3
```

Comment:
I tried this with a cluster (started a 3-node cluster and killed 2 nodes).  This error does not show up in the logs, the query never gets that far.  I see this error on the client:  It will never reach certification.

mysql> insert into t1 values(2, 'nami');
ERROR 1047 (08S01): WSREP has not yet prepared node for application use

Reply:
Reduced the lead-in paragraph to 'Reads are possible until node C decides that it cannot access node A and node B. New writes are forbidden.' & removed the example log output